src/rgb_primaries.rs

```diff
@@ -0,0 +1,101 @@
+//!This module defines the red, blue and green primaries for the comman Rgb color spaces
+use num::Float;
+
+
+use {Yxy, Xyz, Rgb, IntoColor};
+use white_point::{WhitePoint, D65};
+use chromatic_adaptation::AdaptInto;
+use matrix::{Mat3, matrix_inverse, multiply_xyz_to_rgb};
+use flt;
+
+///Represents the tristimulus values for the Rgb primaries
+pub trait RgbPrimaries<Wp, T>
+    where T:Float,
+        Wp: WhitePoint<T>,
+{
+    ///Tristimulus values for red
+    fn red() -> Yxy<Wp, T>;
+    ///Tristimulus values for green
+    fn green() -> Yxy<Wp, T>;
+    ///Tristimulus values for blue
+    fn blue() -> Yxy<Wp, T>;
+}
+
+///Srgb color space with default D65 white point
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct SrgbSpace;
+
+impl<T: Float> RgbPrimaries<D65, T> for SrgbSpace {
+    fn red() -> Yxy<D65, T> {
+        Yxy::new(flt(0.6400), flt(0.3300), flt(0.212656))
+    }
+    fn green() -> Yxy<D65, T> {
+        Yxy::new(flt(0.3000), flt(0.6000), flt(0.715158))
+    }
+    fn blue() -> Yxy<D65, T> {
+        Yxy::new(flt(0.1500), flt(0.0600), flt(0.072186))
+    }
+
+}
+
+///Geneartes to Srgb to Xyz transformation matrix for the given white point
+pub fn rgb_to_xyz_matrix<Wp: WhitePoint<T>, T: Float>() -> Mat3<T> {
+    let r_d65: Xyz<D65, T> = SrgbSpace::red().into_xyz();
+    let g_d65: Xyz<D65, T> = SrgbSpace::green().into_xyz();
+    let b_d65: Xyz<D65, T> = SrgbSpace::blue().into_xyz();
+
+    let wp = Wp::get_xyz();
+    let d65 = D65::get_xyz();
+    let mut transform_matrix = if wp.x == d65.x && wp.y == d65.y && wp.z == d65.z {
+        mat3_from_primaries(r_d65, g_d65, b_d65)
+    } else {
+        let r_wp: Xyz<Wp, T> = r_d65.adapt_into();
+        let g_wp: Xyz<Wp, T> = g_d65.adapt_into();
+        let b_wp: Xyz<Wp, T> = b_d65.adapt_into();
```

Comment:
These adaptions, and this whole `if` should be unnecessary one the previosly mentioned `D65` -> `Wp` change is done.